### PR TITLE
Dropdown: Added 'required' prop support

### PR DESCRIPTION
**Pull request by fantua**
#### Pull request checklist

- [x] Addresses an existing issue: #1188, #1194
- [ ] Include a change request file if publishing 
- [x] New feature
  - [ ] Includes tests
- [x] Documentation update

#### Description of changes

Added `required` prop support. Use of `Label` component for labeling.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -93,7 +94,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     return (
       <div ref='root'>
         { label && (
-          <label id={ id + '-label' } className='ms-Label' ref={ (dropdownLabel) => this._dropdownLabel = dropdownLabel } >{ label }</label>
+          <Label id={ id + '-label' } ref={ (dropdownLabel) => this._dropdownLabel = dropdownLabel } required={required}>{ label }</Label>
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Minor nits:

The `BaseComponent` has a helper for resolving refs:

```tsx
<Label ref={ this._resolveRef('_dropdownLabel') }
```

This helper produces a callback once and then continues to return the same instance so the function isn't regenerated every time render is called.

Also nit spacing inside of the curlies in `{required}`, should be `{ required }`

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -78,7 +79,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
   // Primary Render
   public render() {
     let id = this._id;
-    let { className, label, options, disabled, isDisabled, ariaLabel,
+    let { className, label, options, disabled, isDisabled, ariaLabel, required,
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Can you split these into their own lines:

```ts
let {
  foo,
  bar, 
  etc
} = this.props;

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts [hunk 3]
```diff
@@ -74,6 +74,11 @@ export interface IDropdownProps extends React.Props<Dropdown> {
    */
   disabled?: boolean;
 
+  /**
+   * Whether or not the Dropdown is required.
+   */
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts]: Can you run `rush change` and require a minor fix with the comment:

"Dropdown: Adding a required property which will be passed into the Label."

This is how auto publishing knows what you want to be included in the published build.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -93,7 +101,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     return (
       <div ref='root'>
         { label && (
-          <label id={ id + '-label' } className='ms-Label' ref={ (dropdownLabel) => this._dropdownLabel = dropdownLabel } >{ label }</label>
+          <Label id={ id + '-label' } ref={ this._resolveRef('_dropdownLabel') } required={ required }>{ label }</Label>
```
Comment (rickyp-ms) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: <Label htmlFor={ id }>{ label }</Label>

I think "htmlFor" maybe a better answer than using "id".

Further not sure why you are using ref= _dropdownLabel.  I don't see it being used anywhere.

Comment (rickyp-ms) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: <Label htmlFor={ id }>{ label }</Label>

I think "htmlFor" maybe a better answer than using "id".

Further not sure why you are using ref= _dropdownLabel.  I don't see it being used anywhere.

Comment (rickyp-ms) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Don't add a ref to _dropdownLable if we don't actually reference it anywhere. No need to add code we aren't using.

Comment (fantua) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Yeap, I saw `htmlFor` in your PR, but I think that using `for` attribute is not correct in this case. **Dropdow** is a `div` container and `div` is not _labelable_ element. Use of `for` attr does not give any advantages. See more [here](http://stackoverflow.com/a/11705966).

Comment (rickyp-ms) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Oh good call! I missed the fact that dropdown was a div. Awesome!

Comment (fantua) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: It's was before me, I just added Label component. Probably for some needs it was done :) 
@dzearing, can I remove this without any side-effects?

Comment (scriby) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: It is a div, but it's given the role of combobox, so as far as AX tools see it, it is an input control and not really a div. I wonder if something marked with a role of being an input control is labelable?

**Pull request merged**